src/osd/ReplicatedBackend.cc

```diff
@@ -2180,8 +2234,9 @@ void ReplicatedBackend::trim_pushed_data(
   }
 }
 
-void ReplicatedBackend::_failed_push(pg_shard_t from, const hobject_t &soid)
+void ReplicatedBackend::_failed_pull(pg_shard_t from, const hobject_t &soid)
 {
+  dout(20) << __func__ << ": " << soid << " from " << from << dendl;
   list<pg_shard_t> fl = { from };
   get_parent()->failed_push(fl, soid);
```

Comment:
nevermind, I was confused by the push/pull naming in ReplicatedBackend. having the primary process a push message in _do_pull_response() instead of e.g. _do_push_primary() confused me